DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/DicomUtility.cs

```diff
@@ -0,0 +1,167 @@
+        {
+            EnsureArg.IsNotNull(dateTime, nameof(dateTime));
+
+            // Reference link http://dicom.nema.org/medical/Dicom/2017e/output/chtml/part05/sect_6.2.html
+            Regex dateTimeRegex = new Regex(@"^((?<year>\d{4})(?<month>\d{2})(?<day>\d{2})(?<hour>\d{2})(?<minute>\d{2})(?<second>\d{2})(\.(?<microsecond>\d{1,6}))?(?<timeZone>(?<sign>-|\+)(?<timeZoneHour>\d{2})(?<timeZoneMinute>\d{2}))?)(\s*)$");
+            var matches = dateTimeRegex.Matches(dateTime);
+            if (matches.Count != 1)
+            {
+                throw new DicomDataException("Invalid datetime value. The valid format is YYYYMMDDHHMMSS.FFFFFF&ZZXX.");
+            }
+
+            var groups = matches[0].Groups;
+
+            int year = groups["year"].Success ? int.Parse(groups["year"].Value) : 1;
+            int month = groups["month"].Success ? int.Parse(groups["month"].Value) : 1;
+            int day = groups["day"].Success ? int.Parse(groups["day"].Value) : 1;
+            int hour = groups["hour"].Success ? int.Parse(groups["hour"].Value) : 0;
+            int minute = groups["minute"].Success ? int.Parse(groups["minute"].Value) : 0;
+            int second = groups["second"].Success ? int.Parse(groups["second"].Value) : 0;
+            int millisecond = groups["microsecond"].Success ? int.Parse(groups["microsecond"].Value) / 1000 : 0;
+
+            if (groups["timeZone"].Success)
+            {
+                int sign = int.Parse(groups["sign"].Value + "1");
+                int timeZoneHour = int.Parse(groups["timeZoneHour"].Value) * sign;
+                int timeZoneMinute = int.Parse(groups["timeZoneMinute"].Value) * sign;
+                return new DateTimeObject()
+                {
+                    DateValue = new DateTimeOffset(year, month, day, hour, minute, second, millisecond, new TimeSpan(timeZoneHour, timeZoneMinute, 0)),
+                    HasTimeZone = true,
+                };
+            }
+            else
+            {
+                return new DateTimeObject()
+                {
+                    DateValue = new DateTimeOffset(year, month, day, hour, minute, second, millisecond, default),
+                    HasTimeZone = false,
+                };
+            }
+        }
+
+        public static string GenerateDicomDateString(DateTimeOffset date)
+        {
+            return date.ToString("yyyyMMdd", CultureInfo.InvariantCulture);
+        }
+
+        public static string GenerateDicomDateTimeString(DateTimeObject date)
+        {
+            EnsureArg.IsNotNull(date, nameof(date));
+
+            return date.HasTimeZone == true
+                ? date.DateValue.ToString("yyyyMMddHHmmss.ffffffzzz", CultureInfo.InvariantCulture).Replace(":", string.Empty)
+                : date.DateValue.ToString("yyyyMMddhhmmss.ffffff", CultureInfo.InvariantCulture);
+        }
+
+        public static AgeObject ParseAge(string age)
```

Comment:
🟡 **[Suggestion]**

Below two functions can be moved to `AgeObject` class.
Then the name could be `ParseAgeString` and `ToAgeString`, which makes it clear that these functions handle Age String defined by DICOM standard.